spec/requests/api/job_definitions_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'rails_helper'
+
+describe 'job_definitions' do
+  let(:service_name) { :test_client_name }
+  let(:secret_key) { 'secret_key' }
+  let(:env) do
+    {
+      accept: 'application/json',
+      authorization: "Basic #{Base64.encode64("#{service_name}:#{secret_key}")}",
+    }
+  end
+  let(:result) { JSON.parse(response.body) }
+
+  describe 'GET /v1/definitions' do
+    let(:definition) do
+      create(:job_definition, script: 'noop:', api_allowed: true)
+    end
+
+    before do
+      definition
+    end
+
+    it 'returns definitions' do
+      get "/v1/definitions", params: {}, env: env
+      expect(result).to eq(
+        [{
+          'id' => definition.id,
+          'name' => definition.name,
+          'description' => definition.description,
+          'script' => definition.script
+        }]
+      )
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe 'POST /v1/definitions' do
+    let(:params) do
+      {
+        name: "test",
+        description: "description",
+        script: "noop:",
+        notify_cancellation: 1,
+        hipchat_room: "",
+        hipchat_notify_finished: 1,
+        suspended: false,
+        prevent_multi: 1,
+        hipchat_additional_text: "",
+        text_tags: "",
+        api_allowed: 1,
+        slack_channel: "",
+        webhook_url: "",
+        user_id: [user.id],
+      }
+    end
+
+    let(:user) do
+      create(:user)
+    end
+
+    it 'creates a new definition' do
+      expect {
+        post "/v1/definitions", params: params, env: env
+      }.to change {
+        Kuroko2::JobDefinition.count
+      }.by(1)
+      expect(result['name']).to eq(params[:name])
+      expect(result['description']).to eq(params[:description])
+      expect(result['script']).to eq(params[:script])
+      expect(response.status).to eq(201)
+    end
+  end
+
+  describe 'GET /v1/definitions/:id' do
```

Comment:
I want to test if `job_definition` not found.

Reply:
I have fixed it in 939b69d